sdccc/src/main/java/com/draeger/medical/sdccc/sdcri/testclient/TestClientImpl.java

```diff
@@ -359,17 +401,37 @@ void deviceFound(final ProbedDeviceFoundMessage message) {
         }
 
         sdcRemoteDevice.registerWatchdogObserver(this);
+        isConnected.set(true);
+    }
+
+    @Override
+    public void shouldReconnect(final Boolean shouldReconnect) {
```

Comment:
It is unclear when to call this with false as its parameter value, I think it should be restricted to when all of the following are true: 

1. all log messages of whitelisted threads have been fully processed
2. the maximum amount of retries occurred without a successful one, or, there was a successful try and no further ones will occur
3. the expected behaviour e.g. a disconnect was observed at least once

Perhaps this should be checked here or else some convenience function provided to wait for these events before calling this function.